lib/engine/ability/train_discount.rb

```diff
@@ -18,7 +18,7 @@ def discounted_price(train, price)
 
         discount_value = discount.is_a?(Hash) ? discount[train.name] : discount
 
-        price - (discount_value > 1 ? discount_value : (price * discount_value))
+        price - (discount_value > 1 ? discount_value : (price * discount_value).floor)
```

Comment:
Could this break existing games?

Reply:
I believe all current games use a factor of 10, so division didn't matter. And if it does affect them, they were already broken because JS will round up and Ruby round down (or vice versa).